packages/types/responses.ts

```diff
@@ -16,6 +17,92 @@ export const ZResponsePersonAttributes = ZPersonAttributes.nullable();
 
 export type TResponsePersonAttributes = z.infer<typeof ZResponsePersonAttributes>;
 
+export const ZFilterCriteria = z.object({
+  finished: z.boolean().optional(),
+  createdAt: z
+    .object({
+      min: z.date().optional(),
+      max: z.date().optional(),
+    })
+    .optional(),
+  data: z
```

Comment:
Now with this type for data it's possible to build something like:
```ts
data: {
  op: "lessThan",
  value: ["1","2"],
}
```
which is not valid for this op. For solving this we could build multiple data zod objects for all different `op`s and do a union:
```ts
data: z.union([
  ZResponseFilterCriteriaDataEquals,
  ZResponseFilterCriteriaDataSubmitted,
  ZResponseFilterCriteriaDataLessThan,
  ....
]);

const ZResponseFilterCriteriaDataLessThan = z.object({
  op: z.literal("lessThan"),
  value: z.number(),
});
```